vpr/src/noc/odd_even_routing.h

```diff
@@ -18,25 +18,32 @@
  * in odd columns, while EN and ES turns are not allowed in even columns.
  */
 
-class OddEvenRouting : public TurnModelRouting{
+class OddEvenRouting : public TurnModelRouting {
   public:
+    OddEvenRouting(const NocStorage& noc_model);
+
     ~OddEvenRouting() override;
 
   private:
 
     const std::vector<TurnModelRouting::Direction>& get_legal_directions(NocRouterId src_router_id,
                                                                          NocRouterId curr_router_id,
                                                                          NocRouterId dst_router_id,
+                                                                         TurnModelRouting::Direction prev_dir,
                                                                          const NocStorage& noc_model) override;
 
-    TurnModelRouting::Direction select_next_direction(const std::vector<TurnModelRouting::Direction>& legal_directions,
-                                                      NocRouterId src_router_id,
-                                                      NocRouterId dst_router_id,
-                                                      NocRouterId curr_router_id,
-                                                      NocTrafficFlowId traffic_flow_id,
-                                                      const NocStorage& noc_model) override;
+    bool is_turn_legal(const std::array<std::reference_wrapper<const NocRouter>, 3>& noc_routers,
```

Comment:
Doxygen comment

Reply:
The pure virtual method in the base class (TurnModelRouting) has a doxygen comment. Should I add doxygen comments for the implementations of a pure virtual method in inheriting classes too?

Reply:
If you think the base method comment covers it then you don't need to add another one in the derived class.